src/colmap/estimators/cost_functions.h

```diff
@@ -47,27 +47,83 @@ template <typename T>
 using EigenQuaternionMap = Eigen::Map<const Eigen::Quaternion<T>>;
 using EigenMatrix6d = Eigen::Matrix<double, 6, 6>;
 
+enum CovarianceType {
+  IDENTITY = 1,
+  DIAGONAL = 2,
+  GENERAL = 3,
+};
+
+template <CovarianceType CTYPE = CovarianceType::GENERAL>
 inline Eigen::MatrixXd SqrtInformation(const Eigen::MatrixXd& covariance) {
-  return covariance.inverse().llt().matrixL().transpose();
+  if constexpr (CTYPE == CovarianceType::IDENTITY)
+    return covariance;
+  else
+    return covariance.inverse().llt().matrixL().transpose();
+}
+
+inline CovarianceType GetCovarianceType(const Eigen::MatrixXd& covariance) {
+  if (covariance.isDiagonal()) {
+    if (covariance.isIdentity())
+      return CovarianceType::IDENTITY;
```

Comment:
nit: please put braces also around single line if bodies.

Reply:
Thanks. This is updated.